dali/sasha/saarch.cpp

```diff
@@ -376,6 +376,269 @@ void WUiterate(ISashaCommand *cmd, const char *mask)
+        }
+        void appendOutput(IPropertyTree &wu, const char *outputFormat, StringBuffer &output)
+        {
+            char *saveptr;
+            char *parse = strdup(outputFormat);
+            char *tok = strtok_r(parse, "|,",&saveptr);
+            while (tok)
+            {
+                StringBuffer val;
+                bool found = true;
+                if (strieq(tok, "owner"))
+                    wu.getProp("@submitID", val);
+                else if (strieq(tok, "cluster"))
+                    wu.getProp("@clusterName", val);
+                else if (strieq(tok, "jobname"))
+                    wu.getProp("@jobName",val);
+                else if (strieq(tok,"state"))
+                    wu.getProp("@state", val);
+                else
+                    found = false;
+                if (found)
+                    output.append(',').append(val);
+                tok = strtok_r(NULL, "|,",&saveptr);
+            }
+            free(parse);
+        }
+    public:
+        IMPLEMENT_IINTERFACE_USING(CInterface);
+
+        CWUReader(ISashaCommand *cmd)
+        {
+            owner.set(cmd->queryOwner());
+            cluster.set(cmd->queryCluster());
+            jobName.set(cmd->queryJobName());
+            state.set(cmd->queryState());
+            priority.set(cmd->queryPriority());
+            eclContains.set(cmd->queryEclContains());
+            fileRead.set(cmd->queryFileRead());
+            fileWritten.set(cmd->queryFileWritten());
+            cmdName.set(cmd->queryDfuCmdName());
+
+            dfu = cmd->getDFU();
+
+            //In WUiterate(), the queryAfter() is used as fromDT and the queryBefore() is used as toDT.
+            mkDateCompare(dfu, cmd->queryAfter(), fromDT, '0');
+            mkDateCompare(dfu, cmd->queryBefore(), toDT, '9');
+        }
+        void getWUs(ISashaCommand *cmd, const char *beforeWU, const char *afterWU, unsigned numWUs)
+        {
+            unsigned wuCount = 0;
+            StringArray selectedWUIDs;
+            IArrayOf<IPropertyTree> selectedWUs;
+            const char *outputFormat = cmd->queryOutputFormat();
+            bool checkBeforeOrAfterWU = !isEmptyString(beforeWU) || !isEmptyString(afterWU);
+
+            StringBuffer path;
+            if (dfu)
+                getLdsPath("Archive/DFUWorkUnits", path);
+            else
+                getLdsPath("Archive/WorkUnits", path);
+            Owned<IFile> dir = createIFile(path.str());
+
+            StringBuffer mask, dirMask, fileMask;
+            getMasks(beforeWU, afterWU, mask, dirMask, fileMask);
+            bool sortInc = !isEmptyString(beforeWU) && isEmptyString(afterWU);
+
+            Owned<IRemoteConnection> conn;
+            CIArrayOf<CDirectoryEntry> sortedDirs;
+            Owned<IDirectoryIterator> dirIterator = dir->directoryFiles(dirMask.isEmpty() ? nullptr : dirMask.str(), false, true);
+            sortDirectory(sortedDirs, *dirIterator, SD_bynameNC, sortInc, true); 
```

Comment:
This would be more elegant if implement as part of the dependent issue (HPCC-11841: Create a sorting directory iterator). And would allow anyone to reuse it then.